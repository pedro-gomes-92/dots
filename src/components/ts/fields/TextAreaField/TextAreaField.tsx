import React, { FunctionComponent } from 'react';
import classnames from 'classnames';
import { BaseFieldTypeProps as BaseFieldProps, BaseField } from '../BaseField';
import { TextArea, TextAreaProps } from '../../inputs';

// eslint-disable-next-line @typescript-eslint/no-empty-interface
export interface TextAreaFieldProps extends BaseFieldProps {
  hasFixedSize?: TextAreaProps['hasFixedSize'];
  rows?: TextAreaProps['rows'];
}

export const TextAreaField: FunctionComponent<TextAreaFieldProps> = ({
  model,
  placeHolder,
  value,
  isReadOnly,
  onChange,
  className,
  size,
  rows,
  hasFixedSize,
  ...rest
}: TextAreaFieldProps): JSX.Element => (
  <BaseField className={classnames('is-textarea', className)} {...rest}>
    <TextArea
      model={model}
      value={value}
      onChange={onChange}
      isReadOnly={isReadOnly}
      size={size}
      rows={rows}
      hasFixedSize={hasFixedSize}
      placeHolder={placeHolder}
    />
  </BaseField>
);
