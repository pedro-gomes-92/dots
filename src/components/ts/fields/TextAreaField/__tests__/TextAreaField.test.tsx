import React from 'react';
import { render } from '@testing-library/react';

import { TextAreaField } from '../TextAreaField';

describe('Given <TextAreaField />', (): void => {
  describe('When initializes', (): void => {
    it('Then should match the snapshot', (): void => {
      const { container } = render(<TextAreaField />);
      expect(container).toMatchSnapshot();
    });
  });
});
