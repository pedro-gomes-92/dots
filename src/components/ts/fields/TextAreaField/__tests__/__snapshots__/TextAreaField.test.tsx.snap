// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Given <TextAreaField /> When initializes Then should match the snapshot 1`] = `
<div>
  <div
    class="field is-textarea"
  >
    <div
      class="field-body"
    >
      <div
        class="control"
      >
        <textarea
          class="input textarea"
        />
      </div>
    </div>
  </div>
</div>
`;
